frontend/src/components/Form.jsx

```diff
@@ -0,0 +1,204 @@
+                    id={field.name}
+                    //   key={}
+                    onChange={update("hebrew")}
+                    placeholder={field.name}
+                  />
+                  <input
+                    type="text"
+                    value={formState[field.name]["english"]}
+                    name={field.type}
+                    id={field.name}
+                    //   key={}
+                    onChange={update("english")}
+                    placeholder={field.name}
+                  />
+                </label>
+              </div>
+            );
+          case "string":
+            return (
+              <div>
+                <label>
+                  {field.name}
+                  <input
+                    type="text"
+                    value={formState[field.name]}
+                    // key={}
+                    onChange={update(field.name)}
+                    placeholder={field.name}
+                  />
+                </label>
+              </div>
+            );
+          case "date":
+            return (
+              <div>
+                <label>
+                  {field.name}
+                  <input
+                    type="date"
+                    value={formState[field.name]}
+                    // key={}
+                    onChange={update(field.name)}
+                  />
+                </label>
+              </div>
+            );
+          case "boolean":
+            return (
+              <div>
+                <label>
+                  {field.name}
+                  <input
+                    type="checkbox"
+                    checked={formState[field.name]}
+                    name={field.type}
+                    // key={}
+                    onChange={update(field.name)}
+                  />
+                </label>
+              </div>
+            );
+          case "number":
+            return (
+              <div>
+                <label>
+                  {field.name}
+                  <input
+                    type="number"
+                    value={formState[field.name]}
+                    // key={}
+                    onChange={update(field.name)}
+                    placeholder={field.name}
+                  />
+                </label>
+              </div>
+            );
+          case "image":
+            return (
+              <div>
+                <label>{field.name}
+                <input
+                  type="file"
+                  name="myImage"
+                  onChange={update("image")}
+                  accept="image/*"
+                />
+                <img src={imagePreview} alt="" style={imagestyle}/>
+                </label>
+              </div>
+            );
+          default:
+            break;
+        }
+    }
+
+    return (
+      <form>
+        {createForm()}
+        <button onClick={(e) => handleSubmit(e)}>submit form</button>
+      </form>
+    );
+}
```

Comment:
If a JSX file is more than 150 lines, it's a big red flag and you should break it down to components